oktoberfest/data/spectra.py

```diff
@@ -110,33 +136,37 @@ def add_matrix_from_hdf5(self, intensity_data: pd.DataFrame, fragment_type: Frag
         :param intensity_data: intensity sparse matrix
         :param fragment_type: choose predicted, raw, or mz
         """
-        # generate column names and build dataframe from sparse matrix
-        columns = self._gen_column_names(fragment_type)
-        intensity_data.columns = columns
-        self.add_columns(intensity_data)
+        # add sparse matrix of intensities to corresponding layer
+        if fragment_type.value == 1:
+            self.spectra_data.layers["pred_int"] = scipy.sparse.csr_matrix(intensity_data)
+        elif fragment_type.value == 2:
+            self.spectra_data.layers["raw_int"] = scipy.sparse.csr_matrix(intensity_data)
+        else:
+            self.spectra_data.X = scipy.sparse.csr_matrix(intensity_data)
 
-    def add_matrix(self, intensity_data: pd.Series, fragment_type: FragmentType) -> None:
+    def add_matrix(self, intensity_data, fragment_type: FragmentType, annotation=None) -> None:
         """
         Concatenate intensity df as a sparse matrix to our data.
 
         :param intensity_data: intensity numpy array to add
         :param fragment_type: choose predicted, raw, or mz
         """
-        intensity_df = intensity_data.explode()
-
-        # reshape based on the number of fragments
-        intensity_array = intensity_df.values.astype(np.float32).reshape(-1, c.VEC_LENGTH)
-
+        intensity_array = np.array(intensity_data)
```

Comment:
why do you first ensure nparray, then transform to scipy afterwards? Better add the type annotation Union[csr_matrix, np.ndarray] in the function header and have an
```
if isinstance(intensity_data, np.ndarray): 
   intensity_data = csr_matrix(intensity_data)
```
statement afterwards. This is computationally heavy if it is already csr.

Reply:
I did that, because of the 0 and -1 replacements that are done in the next lines. I checked and it seems like add_matrix is never called with csr_matrices, so I will let this function accept numpy arrays only for now and change the handling accordingly.